### PR TITLE
Feature/arangodoc

### PR DESCRIPTION
**Pull request by kvahed**
Introducing arangoinspect

### CODE REVIEW

**Comment by kvahed** [comment]
usage: start a cluster with authentication. 
`arangoinspect --server.endpoint <transport>://<endpoint>`
type in jwt secret, when prompted

File: arangosh/Shell/V8ClientConnection.cpp [hunk 1]
```diff
@@ -434,15 +434,17 @@ static void ClientConnection_reconnect(
...
-    ConsoleFeature* console =
-        ApplicationServer::getFeature<ConsoleFeature>("Console");
-
-    if (console->isEnabled()) {
-      password = console->readPassword("Please specify a password: ");
-    } else {
-      std::cout << "Please specify a password: " << std::flush;
-      password = ConsoleFeature::readPassword();
-      std::cout << std::endl << std::flush;
+    if (V8ClientConnection::jwtSecret().empty()) {
```
Comment (matthewvon) [inline arangosh/Shell/V8ClientConnection.cpp]: I am curious.  What is the motivation for wrapping this block of code if .empty()?

**Comment by sleto-it** [comment]
Do not merge this, for any reason, until it is extensively documented as per guidelines that will come from @Simran-B 

Thanks,

**Comment by dothebart** [comment]
Please check whether the `scripts/unittest arangosh ` still works with this, and eventually add a test to 
`js/client/modules/@arangodb/testsuites/arangosh.js`. Please note that you may need to have an install'ed copy for all of these tests to succeed, since the shebang-arangosh won't find its js-dir eventually.

**Comment by kvahed** [comment]
i really have no clue, what @dothebart is referring to. the tests ran all along and are green???

**Comment by kvahed** [comment]
`JS_Exit` will not be tested and is yet necessary to exit the shell without user interaction.

**Pull request merged**